### PR TITLE
dev,generate-cgo: externalize cgo files gen into a seperate binary

### PR DESCRIPTION
**Pull request by healthy-pod**
Previously, we only needed to gen cgo files using `dev` so that was were that logic lived. Now, we need to gen cgo files outside dev to fix an issue with gcassert [see #65485].

This code change externalizes the code into a separate binary and let's `dev gen cgo` point to it.

Release note: None
Epic: None

Informs #65485

### CODE REVIEW

File: pkg/cmd/generate-cgo/main.go [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+				}
+				break
+			}
+		}
+	}
+	return ret, nil
+}
+
+func getExecutionRoot() (string, error) {
+	args := []string{"info", "execution_root", "--color=no"}
```
Comment (rickystewart) [inline pkg/cmd/generate-cgo/main.go:95]: Doesn't really make much sense that `workspace` and `bazel-bin` have to be passed in, but `execution_root` does not. I would say just have `generate-cgo` look all of them up by itself.

**Review by rickystewart — APPROVED** [review]

File: pkg/cmd/dev/generate.go [hunk 2]
```diff
@@ -288,70 +278,17 @@ func (d *dev) generateCgo(cmd *cobra.Command) error {
...
-
-	cgoPkgs := []string{
-		"pkg/cli",
-		"pkg/cli/clisqlshell",
-		"pkg/server/status",
-		"pkg/ccl/gssapiccl",
-		"pkg/geo/geoproj",
+	args := []string{
+		"run",
+		"//pkg/cmd/generate-cgo:generate-cgo",
```
Comment (rickystewart) [inline pkg/cmd/dev/generate.go:283]: Should add:

```
"--run_under=cd [workspace] && ",
```

Or something like this to make sure the binary is executed in the workspace directory.

**Review by rickystewart — CHANGES_REQUESTED** [review]

**Review by rickystewart — APPROVED** [review]

**Comment by healthy-pod** [comment]
bors r=rickystewart

**Comment by healthy-pod** [comment]
bors retry

**Comment by healthy-pod** [comment]
bors retry

**Comment by healthy-pod** [comment]
bors single on

**Pull request merged**